### DISCUSSION TITLE
[@types/ramda] may generate d.ts with type reference

### DISCUSSION THREAD

**Discussion by otakustay** [DefinitelyTyped/DefinitelyTyped#58506] [Issues with a @types package]
Assuming this code:

```ts
import {mergeDeepRight} from 'ramda';

export const value = mergeDeepRight({}, {});
```

After `tsc` it's definition (`.d.ts`) becomes:

```ts
/// <reference types="node_modules/ts-toolbelt/out" />
export declare const value: import("Object/Merge").__MergeDeep<{}, {}, 1, import("Misc/BuiltInObject").BuiltInObject, never>;
```

I notice there is a `<reference` statement in `.d.ts` file, isn't it then loose portability because users may not have `ts-toolbelt` at the exact location?

[a comment by an automated account is not shown]

**Comment by Nicholaiii** [comment]
TS Toolbelt is not meant to be installed as a dev dep, so it should be available.

**Comment by otakustay** [comment]
If `@types/ramda` required `ts-toolbelt` on the user side, do we have to define it inside `peerDependencies`?

Interestingly, why it isn't `import {Object} from 'ts-toolbelt';` with `() => Obejct.Merge`, but a strange `<reference>` statement?
